### ISSUE TITLE
Using Pyav for Handling Ffmpeg

### ISSUE THREAD

**Issue by naveen521kk** [ManimCommunity/manim#442]
Currently, AFAIK, Manim calls Ffmpeg using subprocess, which is very slow. In a plan to making manim more faster we can use a Python Ffmpeg wraper, like [Pyav](https://pyav.org/docs/stable/index.html) which is written in C and is a lot faster than directly calling using subprocess and also more Pythonic. I think doing this would just need to edit a file `scene_writer.py`. Also, this would mean that we wouldn't need to ask users to install Ffmpeg as it would be included in PyAv.

### Refs
Pyav Documentation: https://pyav.org/docs/stable/index.html
Pyav Project Source: https://github.com/PyAV-Org/PyAV

**Comment by leotrs** [comment]
Interesting suggestion - I'm generally in favor of this, though I would hesitate to make the change entirely without first having a more robust test suite.

Also, instead of replacing the current file writer to use PyAV, we could implement a new one, FileWriterAV that uses this but keep using the ffmpeg one as default until FileWriterAV is stable.

**Comment by naveen521kk** [comment]
Yeah, @leotrs What you told is good. Let's work on a new class and making it optional.(using config system). But later making it default depreciating the previous one.

**Comment by huguesdevimeux** [comment]
Very intersting idea. Could you PR on this . (if yes, please assign yourself on this issue :D)

**Comment by naveen521kk** [comment]
I think I wouldn't be time to work on this. The issue is open for anyone.

**Comment by manu-chroma** [comment]
Hi, I would like to work on this issue

**Comment by kolibril13** [comment]
Any updates here?
